smarts/core/trap_manager.py

```diff
@@ -45,18 +45,21 @@ class Trap:
     reactivation_time: float
     remaining_time_to_reactivation: float
     patience: float
+    elapsed_sim_time: float = 0
 
-    def step_trigger(self, dt: float):
-        self.remaining_time_to_reactivation -= dt
+    def update_sim_time(self, elapsed_sim_time: float):
+        self.elapsed_sim_time = elapsed_sim_time
 
     @property
     def ready(self):
-        return self.remaining_time_to_reactivation < 0
+        return self.remaining_time_to_reactivation - self.elapsed_sim_time < 0
 
     @property
     def patience_expired(self):
         """Patience recommendation to wait for better capture circumstances"""
-        return self.remaining_time_to_reactivation < -self.patience
+        return (
+            self.remaining_time_to_reactivation - self.elapsed_sim_time < -self.patience
```

Comment:
This bypasses the patience now I think if the starting `elapsed_sim_time` is greater than `patience`.

Reply:
I'll take another look